modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/SdkInternalContext.java

```diff
@@ -46,16 +48,17 @@ public static SdkInternalContext from(CoreEvent event) {
    * SDK components may be nested within each other, so some of the context must be kept separately for the component it belongs
    * to.
    */
-  private final Map<ComponentLocation, LocationSpecificSdkInternalContext> locationSpecificContext = new SmallMap<>();
+  private final Map<Pair<ComponentLocation, String>, LocationSpecificSdkInternalContext> locationSpecificContext =
```

Comment:
Mmmmm... shouldn't a ConcurrentHashMap be faster than a SmallMap synchronized like this?